addons/website_form_project/controllers/main.py

```diff
@@ -2,14 +2,15 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 from odoo.addons.website.controllers import form
+from odoo.http import request
 
 
 class WebsiteForm(form.WebsiteForm):
-    def insert_record(self, request, model, values, custom, meta=None):
-        if model.model == 'project.task':
-            visitor_sudo = request.env['website.visitor']._get_visitor_from_request()
-            visitor_partner = visitor_sudo.partner_id
-            if visitor_partner:
-                values['partner_id'] = visitor_partner.id
-
-        return super().insert_record(request, model, values, custom, meta=meta)
+    def extract_data(self, model, values):
+        res = super().extract_data(model, values)
+        if model.model == 'project.task' and 'email_from' in values:
+            partner_id = request.env['res.partner'].sudo().search([('email', '=', values['email_from'])], limit=1).id
```

Comment:
Got curious passing by.
Anyone can create a task on your behalf and linked to your user/partner just by sending an email without even being logged in?
Was this considered?

Reply:
Seems like easy spam indeed ...

Reply:
@tde-banana-odoo Note that there is a ongoing discussion about that on the task chatter between MAT and AJU, I'll leave that up to you and the related people to decide what should be done, I just raised this "issue" and give my 2 cents so it could at least be discussed by the relevant teams before merge :+1: